cpp/src/bitmask/bit_mask.cuh

```diff
@@ -229,9 +229,9 @@ CUDA_DEVICE_CALLABLE void set_bit_safe(bit_mask_t *valid, T record_idx) {
                 "Record index must be of an integral type");
 
   const gdf_size_type rec{
-      gdf::util::detail::bit_container_index<bit_mask_t, T>(record_idx)};
+      cudf::util::detail::bit_container_index<bit_mask_t, T>(record_idx)};
```

Comment:
I thought the idea behind `detail` namespace is that only functions in `cudf` namespace should be able to access it. So then all of this file's functions should be in `cudf::util::bit_mask::detail`.

Also, this whole header looks like it should be merged with `utilities/bit_util.cuh`

But maybe in another PR.

Reply:
Yeah, we can revisit that stuff.